### PR TITLE
Fixes #15 pause play block status

### LINKED ISSUE #15: Sensor blocks
Most of the existing blocks interact with the actuators. Ideally we should have blocks that provide information about the sensors so that it can be used by "blockliers".

### PR DESCRIPTION
**Pull request by AndriyPt**
This pull request fixes https://github.com/shadow-robot/robot_blockly/issues/15 and fixes https://github.com/shadow-robot/robot_blockly/issues/14

Tested on the latest Google Chrome and Firefox browsers on Ubuntu 14.04.

Functionality can be seen here
play_pause_demo

### CODE REVIEW

**Comment by Nestor94Gonzalez** [comment]
Hi, I tried the code but I get the following error when clicking the pause button. The same happens with launch and resume.
`Traceback (most recent call last):
  File "test.py", line 6, in <module>
    pause
NameError: name 'pause' is not defined
`
Any advice? Thanks

**Comment by AndriyPt** [comment]
@Nestor94Gonzalez can you please give me more output from robot_blockly_backend log? It is printing Python code which it is going to run. Thanks

**Comment by Nestor94Gonzalez** [comment]
@AndriyPt this is what I get when I run just a 'stand up/down' block.
[robot_blockly_backend_LOG.txt](https://github.com/erlerobot/robot_blockly/files/135870/robot_blockly_backend_LOG.txt)

**Comment by AndriyPt** [comment]
@Nestor94Gonzalez From log I can see that you are using old implementation of the robot_blockly_backend.py node. Please clean your build, devel and install folders in your workspace and rebuild whole workspace again using catkin_make.

**Comment by Nestor94Gonzalez** [comment]
Hi @AndriyPt , I've tested the code with some erle-brain blocks I created. I manage to execute the launch/pause/resume properly the first time I run the backend, after that it gets stuck.
This is the log -> [backendLOG.txt](https://github.com/erlerobot/robot_blockly/files/136159/backendLOG.txt)
Line 250 is where the execution fails.

**Comment by AndriyPt** [comment]
Hi @Nestor94Gonzalez 
Thank you for testing this PR. 
I have just merged all changes from master to this PR. 
Among changes was "Save To File" functionality.
Can you please save your workspace to file and attach to this PR.
Also I will need more details from you on what you have done and what went wrong and on which step.
First of all which browser and OS are you using for your testing?
Second, as far as I understand you have done following:

1. Put some connected blocks in the workspace
2. Press "Launch" button
3. Before execution ends you pressed "Pause"
4. Execution stops on the one of the blocks
5. You press "Resume" and let workflow to run till the end
6. You press "Launch" again
7. And? (you was not able to see program execution? Button text was still "Launch"? Selected block doesn't changed?)

Thank you in advance

**Comment by Nestor94Gonzalez** [comment]
Hi @AndriyPt 
I have uploaded the full catkin workspace [here](https://github.com/Nestor94Gonzalez/erle_robot_blockly_pull_req_test.git).
The 'save to file' change is not added here.
**Workspace changes**: status leds blocks added for the erle brain. 
**My PC**: Ubuntu 15.10 & last version of firefox.
**Setup**:
- change frontend/pages/js/execution_logic.js websocket to your ip address. Currently set 192.168.1.50.
- source devel_isolated/setup.bash
- rosrun ros_erle_statusled statusleds.py 
- rosrun robot_blockly robot_blockly_backend.py 

**Test**:
For instance..
blockly_test
**Error**:
It executes as you say from 1-6. Then the log shows '[INFO] [WallTime: 1456133604.854352] Current code status: completed'. 
After this the "Launch" button shows up and sends the code to be executed. Now the "Pause/Resume" is clickable but it does nothing. At this point no block is "selected" and the leds are not turning on neither. So basically only the first time the node is executed works fine.

Thanks for your time

**Comment by AndriyPt** [comment]
@Nestor94Gonzalez I was able to reproduce your issue on my local machine and fixed it.
Also I have merged latest master to this branch.
So you can now use build-in web server for testing.
I have checked functionality on Ubuntu 15.10 latest Firefox and it seems to be working.

File: frontend/pages/js/execution_logic.js [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+  }
+
+  function is_connection_closed() {
+    return (CODE_STATUS.NOT_CONNECTED == current_status);
+  }
+
+  return {
+
+    launch_websockets: function () {
+      socket = new WebSocket("ws://0.0.0.0:9000");
```
Comment (Nestor94Gonzalez) [inline frontend/pages/js/execution_logic.js]: please change this line to:
`
var hostName = window.location.hostname;
socket = new WebSocket("ws://" + hostName + ":9000");
`

**Comment by Nestor94Gonzalez** [comment]
@AndriyPt It is working fine, so it should be merged soon. Thanks!

**Comment by AndriyPt** [comment]
@Nestor94Gonzalez Good news. Thank you too :)

**Pull request merged**